core/modules/ccontrol/UserQueryFactory.cc

```diff
@@ -161,30 +152,28 @@ UserQueryFactory::newUserQuery(std::string const& aQuery,
 // parser maturity will have obviously passed what we could do with antlr v2 and we should remove all the
 // antlr v2 parser code from qserv.
 #if 0
-        // parse using antlr v2
-        std::shared_ptr<query::SelectStmt> a2stmt;
+    {
+        auto parser = parser::SelectParser::newInstance(query, parser::SelectParser::ANTLR2);
```

Comment:
yes, or throw.

Reply:
Oh - sorry, I should probably have looked closer at the code while commenting.
This will not throw, or fail. (provided there's enough memory to make a new object).
`setup` does the work, and may fail via exception.
I don't much like the use pattern in `SelectParser` but it's prior code and I didn't want to change it around any more than I could help, for now at least. The goal here was to conform to existing use patterns in the spirit of making the antlr4 parser interface as much like the old parser interface as possible (in behavior & use)

Reply:
what next: nothing. It's hard to see in the diffs, but this block is actually disabled via `#if 0`, with a comment that explains the reason and intentions for removal.